cargo-afl/src/bin/cargo-afl.rs

```diff
@@ -297,18 +298,59 @@ where
 
     // `-C codegen-units=1` is needed to work around link errors
     // https://github.com/rust-fuzz/afl.rs/pull/193#issuecomment-933550430
+
+    let binding = common::afl_llvm_dir(None);
+    let p = binding.display();
+
     let mut rustflags = format!(
         "-C debug-assertions \
-         -C overflow_checks \
-         -C passes={passes} \
-         -C codegen-units=1 \
-         -C llvm-args=-sanitizer-coverage-level=3 \
-         -C llvm-args=-sanitizer-coverage-trace-pc-guard \
-         -C llvm-args=-sanitizer-coverage-prune-blocks=0 \
-         -C llvm-args=-sanitizer-coverage-trace-compares \
-         -C opt-level=3 \
-         -C target-cpu=native "
+             -C overflow_checks \
+             -C passes={passes} \
+             -C codegen-units=1 \
+             -C opt-level=3 \
+             -C target-cpu=native "
     );
+    let mut environment_variables = HashMap::<String, String>::new();
+    environment_variables.insert("ASAN_OPTIONS".to_string(), asan_options);
+    environment_variables.insert("TSAN_OPTIONS".to_string(), tsan_options);
+
+    if cfg!(feature = "cmplog") {
+        // Make sure we are on nightly for the -Z flags
+        assert!(
+            rustc_version::version_meta().unwrap().channel == rustc_version::Channel::Nightly,
+            "cargo-afl must be compiled with nightly for the cmplog feature"
+        );
+
+        let llvm_config = common::get_llvm_config();
+
+        // check if llvm tools are installed and with the good version for the plugin compilation
+        let mut command = Command::new(llvm_config.clone());
+        command.args(["--version"]);
+        let status = command
+            .status()
+            .unwrap_or_else(|_| panic!("could not run {llvm_config} --version"));
+        assert!(status.success());
+
+        rustflags.push_str(&format!(
+            "-Z llvm-plugins={p}/cmplog-instructions-pass.so  \
+            -Z llvm-plugins={p}/cmplog-routines-pass.so \
+            -Z llvm-plugins={p}/cmplog-switches-pass.so \
+            -Z llvm-plugins={p}/SanitizerCoveragePCGUARD.so
+             "
+        ));
```

Comment:
For my understanding, why is this list of `so`s less than those listed in `build_afl_llvm_plugins`?

Reply:
We were thinking to add some functionality in the future that depends on these other plugins, thus included all of them in the copied directory